src/EFCore.PG/Query/EvaluatableExpressionFilters/Internal/NpgsqlNodaTimeEvaluatableExpressionFilter.cs

```diff
@@ -24,56 +24,22 @@
-        bool IEvaluatableExpressionFilter.IsEvaluatableConstant(ConstantExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableElementInit(ElementInit node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableInvocation(InvocationExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableLambda(LambdaExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableListInit(ListInitExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableMemberAssignment(MemberAssignment node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableMemberInit(MemberInitExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableMemberListBinding(MemberListBinding node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableMemberMemberBinding(MemberMemberBinding node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableNew(NewExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableNewArray(NewArrayExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableTypeBinary(TypeBinaryExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableUnary(UnaryExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableBlock(BlockExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableCatchBlock(CatchBlock node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableDebugInfo(DebugInfoExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableDefault(DefaultExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableGoto(GotoExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableIndex(IndexExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableLabel(LabelExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableLabelTarget(LabelTarget node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableLoop(LoopExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableSwitch(SwitchExpression node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableSwitchCase(SwitchCase node) => true;
-        bool IEvaluatableExpressionFilter.IsEvaluatableTry(TryExpression node) => true;
-
-        #endregion
+        public override bool IsEvaluatableExpression(Expression expression)
+            => !(
+                (expression is MethodCallExpression methodExpr &&
+                methodExpr.Method.DeclaringType?.FullName == "NodaTime.SystemClock" &&
+                methodExpr.Method.Name == "GetCurrentInstant")
+                ||
+                (expression is MemberExpression memberExpr &&
+                memberExpr.Member.DeclaringType?.FullName == "NodaTime.SystemClock" &&
+                memberExpr.Member.Name == "Instance"));
```

Comment:
It's out of scope for this PR, but is there a reason we aren't storing static fields of the `MethodInfo` and `PropertyInfo` for these comparisons?

Reply:
Yep, there is :) These are NodaTime symbols. The NodaTime plugin is the one actually referencing the NodaTime assembly, but since there's no plugin-based evaluatable filter option, we have to do this in the core which doesn't/shouldn't reference NodaTime. See https://github.com/aspnet/EntityFrameworkCore/issues/13454 (this is why we have the TODO above.